README.md

```diff
@@ -2,55 +2,256 @@
+host in the `hosts` list under `vpn_connections`.  This gives the admin flexibility in case they cannot edit values in one or the other section.
+
+```yaml
+all:
+  hosts:
+    bastion_east:
+      ansible_host: bastion1.example.com # the hostname that Ansible uses
+      vpn_host: 192.168.122.101 # the IP address we want to use for the tunnel
+    bastion_west:
+      ansible_host: bastion2.example.com
+      vpn_host: 192.168.122.102
+    bastion_north:
+      ansible_host: bastion3.example.com
+  vars:
+    vpn_connections:
+      - hosts:
+          bastion_east:
+          bastion_west:
+          bastion_north:
+            vpn_host: 192.168.122.103
+```
+
+## Use Cases
+
+- Host-to-Host (openstack): Specific nodes connecting to each other. Use IPsec for IP failover between these nodes (so all other  
+nodes don&#39;t need to be aware of anything happening). Keys are FreeIPA certificates, and pre-shared keys
+- Host-to-Host (data centers): Two systems in different data centers communicate encrypted with each other using FreeIPA  
+certificates, and pre-shared keys (open question: rsa keys)
+- Host-to-Host (one host): One system communicating with an existing system (e.g., cisco) in an other organization that uses  
+pre-shared keys
+- Network-to-Network (two routers): One organization router connecting to a second one bringing together two distinct networks.  
+Keys are FreeIPA certificates, and pre-shared keys.
+- VPN Remote Access Server / Roadwarrior: One organization router accepting connections from multiple clients. Clients connect to  
+a single router using FreeIPA certificates.
+- MESH: node independent configurations. When adding/removing a node, you don&#39;t need to reconfigure all other nodes. They all  
+attempt to setup individual host-to-host connections. A PKI is used to authenticate nodes (FreeIPA, potentially in the future DNSSEC)
+
+
+For a couple of these use cases, you cannot use per-host settings in `all.hosts`.
+
+### Host-to-host (one host)
+
+In this case, one of the hosts is external to the inventory e.g. in a remote datacenter, and you can only set up the local ends of  
+the tunnels.
+
+```yaml
+all:
+  hosts:
+    bastion_east:
+      ansible_host: bastion1.example.com
+    bastion_west:
+      ansible_host: bastion2.example.com
+  vars:
+    vpn_connections:
+      - hosts:
+          bastion_east:
+          bastion_west:
+          bastion_north: # not in the hosts list
+            vpn_host: 192.168.122.103
+            external: true
+```
+
+By setting `external: true` means that we know `bastion_north` is not in the Ansible inventory, so do not warn that this host is unknown.  
+Or perhaps we can assume that if the host is not in the list, but sufficient parameters are specified, the host is external, and do not warn.  
+We might also use `external: true` as a hint that verification can fail if the remote end of the tunnel is not yet set up.
```

Comment:
We'll have to revisit this once you get into the implementation details, to see if we need this or not.

Reply:
Not sure where we should document it - maybe here, maybe elsewhere - to say something to the effect of - if you specify all of the host parameters with the host, for any of the use cases, then you don't have to use the Ansible inventory name for the host, you can use any unique, descriptive name - however, it is still recommended to use the name from the Ansible inventory, if possible - the role can check that the name is valid.

Reply:
Revisiting this issue - I don't think the `external` parameter is explicitly necessary. The alternative is that we assume that if a host is specified in a tunnel configuration and it is not in the inventory list of hosts, it must be external and therefore a hostname/IP address must be specified (otherwise the playbook fails). I was trying to think of any edge cases where this would be a wrong assumption and nothing comes to mind.

Reply:
> The alternative is that we assume that if a host is specified in a tunnel configuration and it is not in the inventory list of hosts, it must be external and therefore a hostname/IP address must be specified (otherwise the playbook fails)

The case for using `external` is if a user makes a typo in the tunnel configuration e.g.
```
hosts:
  a.exmaple.com:
...
```
The role will give an error e.g. `a.exmaple.com not found` when the user meant to say `a.example.com`.  If the user provides `external: true` in this case because `a.exmaple.com` is a real hostname, then the role will not give an error.

But - for now, let's go with your suggestion and omit `external`, and see if this is actually a problem for real customers.